mvvm/src/main/java/com/thefuntasty/mvvm/ViewModelCreator.kt

```diff
@@ -0,0 +1,11 @@
+package com.thefuntasty.mvvm
+
+import com.thefuntasty.mvvm.factory.BaseViewModelFactory
+import kotlin.reflect.KClass
+
+interface ViewModelCreator<VM : BaseViewModel<*>> {
+
+    val viewModelFactory: BaseViewModelFactory<VM>
+
+    val viewModelClass: KClass<VM>
```

Comment:
Co kdyby jsme toto "schovali" do `BaseViewModelFactory`, aby jsi to musel implementovat v konkrétní factory viewmodelu (kterou stejně generujeme) a nebylo to na očích v aktivitě / fragmentu ?

Reply:
@voneskuba Podla mna je to asi jedno, stejne to bude generovane aj v activity/fragment

Reply:
Na druhou stranu, pomocí té factory vytváříš viewmodel a pokud potřebuješ dodatečnou informaci, aby jsi viewmodel správně uložil do store, asi by ju měla znát factory a ne UI. Ale jsem přikloněn k obojímu.

Reply:
Jinak je to super :-)

Reply:
Very nice idea!

Reply:
Fixed @voneskuba @matejcikm